node/node_v2.go

```diff
@@ -0,0 +1,126 @@
+// These v2 methods are implemented in this separate file to keep the code organized.
+// Note that there is no NodeV2 type and these methods are implemented in the existing Node type.
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/gammazero/workerpool"
+)
+
+type requestMetadata struct {
+	blobShardIndex int
+	quorum         core.QuorumID
+}
+type relayRequest struct {
+	chunkRequests []*clients.ChunkRequestByRange
+	metadata      []*requestMetadata
```

Comment:
This looks not metadata, but an integral part of the request itself (without quorum ID, the request is incomplete). That is, a well defined request should have blobkey, chunk index range and quorum ID all together.

Reply:
This is metadata for keeping track of the requests, i.e. the actual request to relay doesn't require quorum IDs, just blob key and indices. This metadata is used on the client side (node) to put the responses in the right place.